conversion.go

```diff
@@ -565,7 +566,25 @@ func (dst *Int) Scan(src interface{}) error {
 	}
 	switch src := src.(type) {
 	case string:
-		return dst.SetFromDecimal(src)
+		splt := strings.SplitN(src, "e", 2)
+		if len(splt) < 2 {
+			return dst.SetFromDecimal(src)
+		}
+		err := dst.SetFromDecimal(splt[0])
+		if err != nil {
+			return err
+		}
+		if splt[1] == "0" {
+			return nil
+		}
+		exp := new(Int)
+		err = exp.SetFromDecimal(splt[1])
+		if err != nil {
+			return err
+		}
+		exp.Exp(NewInt(10), exp)
+		dst.Mul(dst, exp)
```

Comment:
For `Exp` we have no overflow checks, 
```
// Exp sets z = base**exponent mod 2**256, and returns z.
```

For `Mul`, we do have `MulOverflow`, so we could check that and return error if it does not fit. But it feels a bit wonky to just silently fail the conversion (because of the `Exp`) without signalling an error. 

IMO, it would have been ok if the failure-mode had been: 
- The result is `X mod 2^256`, where `X` is the correct result. 

But I think the possible failures are different .. (?)

Reply:
i think you are right here, they will be different.

i think the popular postgresql scanner will error anyways if the value is too large to be scanned in, so I think that it is fine to do the same.